packages/dom/src/layout/mod.rs

```diff
@@ -144,10 +144,36 @@ impl LayoutPartialTree for Document {
 
                     // todo: need to handle shadow roots by actually descending into them
                     if *element_data.name.local == *"input" {
-                        // if the input type is hidden, hide it
-                        if let Some("hidden") = element_data.attr(local_name!("type")) {
-                            node.style.display = Display::None;
-                            return taffy::LayoutOutput::HIDDEN;
+                        match element_data.attr(local_name!("type")) {
+                            // if the input type is hidden, hide it
+                            Some("hidden") => {
+                                node.style.display = Display::None;
+                                return taffy::LayoutOutput::HIDDEN;
+                            }
+                            Some("checkbox") => {
+                                return compute_leaf_layout(
+                                    inputs,
+                                    &node.style,
+                                    |_known_size, _available_space| {
```

Comment:
I suspect that the vertical alignment issue you are seeing is due to taffy/parley assuming the bottom of the node's border-box as it's baseline. Whereas for checkboxes it probably ought to ignore the margin and use the bottom of the actual checkbox as the baseline.

We will need to add support for inline-box baselines to parley to make that work properly, but a good first step would be to modify the `first_baselines.y` field of the `taffy::LayoutOutput` returned by `compute_leaf_layout` to set the baseline to the bottom of the checkbox (height of checkbox + top padding/border/margin).